OSX_INSTALL.md

```diff
@@ -0,0 +1,56 @@
+# Installation on Mac OS Sierra (10.12)
+
+## NOTE!! Currently, these instructions do not result in a functional build.
+
+Install **Homebrew** (if not already installed)
+```
+/usr/bin/ruby -e "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install)"
+```
+
+Install build tools: **LLVM**, **GCC**, **CMAKE**, and **GIT**
+```
+brew install llvm gcc cmake git
+```
+
+## Verify Installation (Optional)
+
+These not necessary but useful if you want to verify the above installation was successful.
+
+```which clang``` should be ```/usr/local/opt/llvm/bin/clang```
+
+```which clang++``` should be ```/usr/local/opt/llvm/bin/clang++```
+
+```which gcc-7``` should be ```/usr/local/bin/gcc-7```
+
+```which cmake``` should be ```/usr/local/bin/cmake```
+
+## Add Build Tools to Your Environment
+
+Temporarily add build tools to your environment (for the current terminal session)
+```
+export PATH="/usr/local/opt/llvm/bin:$PATH" && export CC=clang && export CXX=clang++
+```
+
+Permanently add build tools to your environment
+```
+echo 'export PATH="/usr/local/opt/llvm/bin:$PATH" && export CC=clang && export CXX=clang++' >> ~/.bash_profile
+```
+
+## Set Build Flags
+
+```
+export CXXFLAGS=$(./flags.sh cxx) && export LDFLAGS=$(./flags.sh ld)
```

Comment:
how about merging this into one `cmake-osx.sh` script, which sets the flags and runs cmake?
```bash
#!/bin/bash
export CXXFLAGS=$(./flags.sh cxx) && export LDFLAGS=$(./flags.sh ld)
cmake $@
```
Or just adding a tip at the bottom to put all that in a `build.sh` script

Reply:
everything i read says you can't set flags from a shell script.

Reply:
well you can't, the variables are only set inside that shell script, but if that same script also runs the `cmake` command, thats the only place the variables need to be set. the `$@` passes all arguments on to cmake

Reply:
sweeet!  so then we can set ```CC``` and ```CXX``` along with the ```CXXFLAGS``` and trigger cmake all in one script.  I'll fix it up and push later today.

Reply:
yep!